File: LENTABP10/src/main/java/com/lenta/lentabp10/models/repositories/ITaskWriteOffReasonRepository.java
package com.lenta.lentabp10.models.repositories;

import com.lenta.lentabp10.models.task.TaskWriteOffReason;
import com.lenta.shared.models.core.IProduct;

import java.util.List;

public interface ITaskWriteOffReasonRepository {
    List<TaskWriteOffReason> findWriteOffReasonsOfProduct(IProduct product);

    boolean addWriteOffReason(TaskWriteOffReason writeOffReason);
    boolean deleteWriteOffReason(TaskWriteOffReason writeOffReason);
    boolean deleteWriteOffReasonsForProduct(IProduct product);
    void clear();

    TaskWriteOffReason get(int index);
    int lenght();
}

File: writeoff_breakage/src/main/java/com/lenta/bp10/models/repositories/ITaskWriteOffReasonRepository.kt
package com.lenta.bp10.models.repositories

import com.lenta.bp10.models.task.TaskWriteOffReason
import com.lenta.shared.models.core.ProductInfo

interface ITaskWriteOffReasonRepository {
    fun getWriteOffReasons(): List<TaskWriteOffReason>
    fun findWriteOffReasonsOfProduct(product: ProductInfo): List<TaskWriteOffReason>
    fun addWriteOffReason(writeOffReason: TaskWriteOffReason): Boolean
    fun deleteWriteOffReason(writeOffReason: TaskWriteOffReason): Boolean
    fun deleteWriteOffReasonsForProduct(product: ProductInfo): Boolean
    fun clear()
    fun get(index: Int): TaskWriteOffReason
    fun lenght(): Int
}